packages/gatsby/src/services/calculate-dirty-queries.ts

```diff
@@ -0,0 +1,19 @@
+import { calcInitialDirtyQueryIds, groupQueryIds } from "../query"
+import { IBuildContext } from "./"
+
+export interface IGroupedQueryIds {
+  pageQueryIds: string[]
+  staticQueryIds: string[]
+}
+
+export async function calculateDirtyQueries({
+  store,
+}: Partial<IBuildContext>): Promise<{
+  queryIds: IGroupedQueryIds
+}> {
+  const state = store?.getState()
+  // TODO: Check filesDirty from context
+
+  const queryIds = calcInitialDirtyQueryIds(state)
+  return { queryIds: groupQueryIds(queryIds) }
```

Comment:
Unsure if we actually want to nest this into queryIds object and not just returning groupQueryIds directly